ckanext-datagov_inventory/ckanext/datagov_inventory/plugin.py

```diff
@@ -0,0 +1,213 @@
+def group_list(context, data_dict=None):
+    log.info('Calling group list')
+    return validate_user(context, data_dict)
+
+
+# group list is called on anonymous pages through package_show
+# this prevents a 500, and instead returns our 403 error
+@plugins.toolkit.auth_allow_anonymous_access
+def group_list_authz(context, data_dict=None):
+    log.info('Calling group list authz')
+    user = context.get('user')
+    if user:
+        return authz.is_authorized('group_list', context, data_dict)
+    else:
+        toolkit.abort(status_code=403)
+        return {'success': False, 'msg': default_message}
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def license_list(context, data_dict=None):
+    log.info('Calling license list')
+    return validate_user(context, data_dict)
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def member_roles_list(context, data_dict=None):
+    log.info('Calling member roles list')
+    return validate_user(context, data_dict)
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def organization_list(context, data_dict=None):
+    log.info('Calling organization list')
+    return validate_user(context, data_dict)
+
+
+# this is already handled non-auth ckan core requests
+# implementing this here causes errors in other functions calls
+# ie package_search
+# @plugins.toolkit.auth_disallow_anonymous_access
+# def organization_list_for_user(context, data_dict=None):
+#     log.info('Calling Organization List For User')
+#     return validate_user(context, data_dict)
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def package_search(context, data_dict=None):
+    log.info('Calling package search')
+    return validate_user(context, data_dict)
+
+
+# Let upstream CKAN handle package_show access normally,
+#   depending upon the user
+# @plugins.toolkit.auth_allow_anonymous_access
+# def package_show(context, data_dict):
+#     return {'success' : True}
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def package_list(context, data_dict=None):
+    log.info('Calling package list')
+    return validate_user(context, data_dict)
+
+
+# resources should be the only 'action' that defaults to True
+@plugins.toolkit.auth_allow_anonymous_access
+def resource_show(context, data_dict=None):
+    log.info('Calling resource show')
+    return {'success': True}
+
+
+@plugins.toolkit.auth_allow_anonymous_access
+def resource_read(context, data_dict=None):
+    log.info('Calling resource read')
+    return {'success': True}
+
+
+@plugins.toolkit.auth_allow_anonymous_access
+def resource_view_list(context, data_dict=None):
+    log.info('Calling resource view list')
+    return {'success': True}
+
+
+@plugins.toolkit.auth_allow_anonymous_access
+def resource_view_show(context, data_dict=None):
+    log.info('Calling resource view')
+    return {'success': True}
+
+
+@plugins.toolkit.auth_disallow_anonymous_access
+def request_reset(context, data_dict=None):
+    # Using login.gov: reject all CKAN password resets
```

Comment:
@adborden Is there a use case for inventory where we need/want users to be able to reset their password? It looks like the [saml2auth extension doesn't cover this case](https://github.com/keitaroinc/ckanext-saml2auth/search?q=request_reset), and I want to make sure nothing can be done to the user that would cause them to be locked out of their account...

Reply:
And at least for now, yes. We're currently using local CKAN users in sandbox for data.gov contractors without PIV cards and it looks like we're going to do that in production as well. saml and local users can co-exist with the saml2auth extension.

Also, there is a use case for [service accounts](https://github.com/GSA/datagov-deploy/issues/2634) which look like they would be implemented as local CKAN users. If we want them to be self-serve for agencies, we should allow password resets.